core/input/mouse.js

```diff
@@ -5,59 +5,57 @@
  * Licensed under MPL 2.0 or any later version (see LICENSE.txt)
  */
 
-/*jslint browser: true, white: false */
-/*global window, Util */
-
 import * as Log from '../util/logging.js';
 import { isTouchDevice } from '../util/browsers.js';
 import { setCapture, stopEvent, getPointerEvent } from '../util/events.js';
 
-var WHEEL_STEP = 10; // Delta threshold for a mouse wheel step
-var WHEEL_STEP_TIMEOUT = 50; // ms
-var WHEEL_LINE_HEIGHT = 19;
+const WHEEL_STEP = 10; // Delta threshold for a mouse wheel step
+const WHEEL_STEP_TIMEOUT = 50; // ms
+const WHEEL_LINE_HEIGHT = 19;
 
-export default function Mouse(target) {
-    this._target = target || document;
+export default class Mouse {
+    constructor(target) {
+        this._target = target || document;
 
-    this._doubleClickTimer = null;
-    this._lastTouchPos = null;
+        this._doubleClickTimer = null;
+        this._lastTouchPos = null;
 
-    this._pos = null;
-    this._wheelStepXTimer = null;
-    this._wheelStepYTimer = null;
-    this._accumulatedWheelDeltaX = 0;
-    this._accumulatedWheelDeltaY = 0;
+        this._pos = null;
+        this._wheelStepXTimer = null;
+        this._wheelStepYTimer = null;
+        this._accumulatedWheelDeltaX = 0;
+        this._accumulatedWheelDeltaY = 0;
 
-    this._eventHandlers = {
-        'mousedown': this._handleMouseDown.bind(this),
-        'mouseup': this._handleMouseUp.bind(this),
-        'mousemove': this._handleMouseMove.bind(this),
-        'mousewheel': this._handleMouseWheel.bind(this),
-        'mousedisable': this._handleMouseDisable.bind(this)
-    };
-};
+        this._eventHandlers = {
+            'mousedown': this._handleMouseDown.bind(this),
+            'mouseup': this._handleMouseUp.bind(this),
+            'mousemove': this._handleMouseMove.bind(this),
+            'mousewheel': this._handleMouseWheel.bind(this),
+            'mousedisable': this._handleMouseDisable.bind(this)
+        };
 
-Mouse.prototype = {
     // ===== PROPERTIES =====
 
-    touchButton: 1,                 // Button mask (1, 2, 4) for touch devices (0 means ignore clicks)
+    this.touchButton = 1;                 // Button mask (1, 2, 4) for touch devices (0 means ignore clicks)
```

Comment:
This did not convert nicely.

Reply:
Yup. Declaring variables as `touchButton = 1` as you can do with methods is ES7 so it would need transpilation to all browsers.

So the way to declare value properties in ES6 is to declare them like this in the constructor.
I've put them always at the bottom so it should be trivial to take them out once ES7 is more widespread.

Reply:
Urgh. Are there any other patterns that can be used? I'd like to avoid the confusion of putting it in the constructor with other stuff. And for `RFB` it gets extra messy as it has getters and setters for some properties but not all.

Reply:
I think that we could use https://babeljs.io/docs/plugins/transform-class-properties/
```
class MyClass {
    prop = true

    function() { ... }
}
```

However we would need to transpile it for all browsers AFAIK. It's not even part of es2017 preset.

Reply:
That's a no-go in that case. We need something that everyone (except IE) supports.

Reply:
Agreed. That's why I opted for adding them at the bottom of the constructor with a comment stating that they are properties. Once the class properties become a standard, it's trivial to move them out of there. :)